modules/tinymce/src/plugins/accordion/main/ts/core/Actions.ts

```diff
@@ -29,4 +32,57 @@ const insertAccordion = (editor: Editor): void => {
   }
 };
 
-export { insertAccordion };
+const toggleDetailsElement = (details: HTMLDetailsElement, state?: boolean): boolean => {
+  const shouldOpen = state ?? !Utils.isAccordionOpen(details);
+  if (shouldOpen) {
+    details.setAttribute('open', 'open');
+    details.setAttribute('data-mce-open', 'open');
+  } else {
+    details.removeAttribute('open');
+    details.removeAttribute('data-mce-open');
+  }
+  return shouldOpen;
+};
+
+const toggleAccordion = (editor: Editor, state?: boolean): void => {
+  const details = Utils.getSelectedDetails(editor);
+  if (details) {
+    Events.fireToggleAccordionEvent(editor, details, toggleDetailsElement(details, state));
+  }
+};
+
+const removeAccordion = (editor: Editor): void => {
+  const details = Utils.getSelectedDetails(editor);
+  if (!details) {
+    return;
+  }
+
+  const { nextSibling } = details;
+  if (nextSibling) {
+    editor.selection.setCursorLocation(nextSibling, 0);
```

Comment:
What happens if there is not nextSibling where does the selection go. Is (nextSibling, 0) always a valid spot? What if the next sibling is a table. I don't think `<table>|<tbody>` is a valid spot. You could maybe use:
```
editor.selection.select(nextSibling, true);
editor.selection.collapse(true);
```
That will find the first text node or BR spot then move the selection to that and then collapse. It would be good if we had better public apis to do this find the first suitable spot for a node.

Reply:
Works great, thanks!